src/selection.ts

```diff
@@ -1,5 +1,33 @@
 import dom from './dom';
 
+const isIframe = (node: Node) => node.nodeName === 'IFRAME';
+const isImg = (node: Node) => node.nodeName === 'IMG';
+
+export const cleanSelection = (selection: Selection): Selection => {
+  const anchor = selection.anchorNode;
+  const focus = selection.focusNode;
+
+  if (!anchor || !focus) {
+    return selection;
+  }
+
+  const anchorParent = anchor.parentNode;
+  const focusParent = focus.parentNode;
+  const beginsOnIframe = isIframe(anchor);
+  const endsOnIframe = isIframe(focus);
+
+  if (beginsOnIframe && anchorParent && !endsOnIframe) {
```

Comment:
idk what really this is doing, but it occurs to me that it might matter if its _the same_ iframe, vs a different one, right?

Reply:
No, it shouldn't matter—this is just ensuring that the focus stays the same. Will try to reorganize a bit for clarity.